### ISSUE TITLE
Map rotation

### ISSUE THREAD

**Issue by liuzy1993** [osmdroid/osmdroid#1456]
It can automatically rotate the map when driving according to the route, so that my direction is facing the direction of the road

**Comment by spyhunter99** [comment]
yes, this was asked recently. there is a sample called heading/compass up

**Comment by liuzy1993** [comment]
I don't use GPS positioning now, simulate the movement of GPS points, and then make my customized forward icon, can I achieve it?

**Comment by spyhunter99** [comment]
https://github.com/osmdroid/osmdroid/blob/master/OpenStreetMapViewer/src/main/java/org/osmdroid/samplefragments/location/SampleHeadingCompassUp.java

basically on each map update, move the icon and center the map on the new location, updating the rotation if needed

**Comment by liuzy1993** [comment]
![ss2](https://user-images.githubusercontent.com/13444860/71330928-8f59a780-256a-11ea-9bb5-3dee870c785b.gif)
This is the effect I want to achieve. Make sure that my icon arrow is always upward. Can your code be implemented?

**Comment by RDX-Awais** [comment]
i also found this solution. its work on onRotate function but giving the right angle how to find right angle ?
